client/src/modules/depots/modals/depot.modal.html

```diff
@@ -99,6 +99,24 @@
       </div>
     </div>
 
+    <div class="form-group">
+      <div class="checkbox">
+        <label>
+          <input
+            type="checkbox"
+            name="join_location"
+            ng-true-value="1"
+            ng-false-value="0"
+            ng-model="DepotModalCtrl.depot.joinLocation">
```

Comment:
It looks like `joinLocation` is just a flag for the view to show/hide the depot's location select.  Therefore, you don't need to attach it to the `depot` object - this causes you to reference it on the server.

I propose that you make another variable such as this:
```js
// this will be "true" if the uuid is set
vm.hasDepotLocation = !!(vm.depot.location_uuid);
```
Then you can just toggle `hasDepotLocation` without changing properties on `vm.depot`.

If on the server you need to know if it has a location, you can do the same thing!

Reply:
the `joinLocation` flag is required for depot on the server side if we want to remove a location on a depot